_posts/2018/2018-06-18-bridging-the-gap-between-literate-programming-and-teaching-modules.markdown

```diff
@@ -0,0 +1,94 @@
+---
+layout: "post"
+title: "Bridging the gap between literate programming and teaching modules"
+date: "2018-06-18 11:07"
+author: Tania Allard
+tags:
+  - use-case
+  - best-practice
+  - open-science
+  - teaching
+  - jupyter
+---
+
+## Background
+The Jupyter notebook is a popular tool used by thousands of individuals and organisations all over the world. The notebook popularity is associated
+with the paradigm of **literate programming**, first introduced by
+**Donald E. Knuth**. Under such a paradigm the (computer) program is treated like
+a piece of literature addressed to humans rather than to computers.
+As a consequence, such programs focus on the logic and flow of human thought.
+At the same time, the programs integrate both the data analysis and textual documentation,
+making the programs more robust, more portable, and easier to maintain than
+programs are solely written in high-level languages.
+
+The notebooks or rather the Jupyter Ecosystem have found their place within research,
+education and data science, among many other areas, particularly promoting
+knowledge transfer. I first started using the notebooks when I realized that they
+allowed me not to only write code in a number of programming languages but also
+allowed me to build a compelling narrative in which analysis, documentation, and data are
+presented together transparently. Since then I have used the notebooks
+for dissemination, education, prototyping, have developed ad-hoc extensions/widgets and
+had helped others to use the notebooks better.
+
+However, the emergence of novel technologies also brings a new set of challenges
+to overcome during their implementation and usage. This case study presents
+some of the work the members of ODK have done in order to bring literate programming to
+the classrooms.
+
+## Scenario
+Most universities in the UK have an institutional learning platform commonly used
+by lecturers to share materials with students.
+Such a platform only allows for upload and rendering of documents in .pdf, .txt,
```

Comment:
Such a platform only allows for upload and rendering of **static** documents in .pdf, .txt,